docs/DesignDocs/CryptoLibConfiguration.md

```diff
@@ -0,0 +1,115 @@
+:---|:---|:---|
+MBEDTLS_HAVE_ASM | !OPENSSL_NO_ASM | OpenSSL enables ASM by default (MD5_ASM, SHA1_ASM, RMD160_ASM, SHA256_ASM, SHA512_ASM, AES_ASM).  |
+MBEDTLS_HAVE_SSE2 | OPENSSL_IA32_SSE2 | Enabled since SGX chips should all support SSE2. OpenSSL enables sse2 by default. |
+MBEDTLS_CIPHER_MODE_CBC | N/A | OpenSSL supports CBC mode by default. |
+!MBEDTLS_CIPHER_MODE_CFB | N/A | OpenSSL supports CFB mode by default. |
+MBEDTLS_CIPHER_MODE_CTR | N/A | OpenSSL supports CTR mode by default. |
+MBEDTLS_CIPHER_MODE_OFB | N/A | OpenSSL supports OFB mode by default. |
+MBEDTLS_CIPHER_MODE_XTS | N/A | OpenSSL supports XTS mode by default. |
+!MBEDTLS_CIPHER_NULL_CIPHER | N/A | OpenSSL does not have equivalent option. Mbed disables this option along with !MBEDTLS_ENABLE_WEAK_CIPHERSUITES. |
+!MBEDTLS_ENABLE_WEAK_CIPHERSUITES | OPENSSL_NO_WEAK_SSL_CIPHERS | Both implementations disable a similar set of ciphersuites. Require further investigation to check the difference. |
+MBEDTLS_CIPHER_PADDING_PKCS7 | N/A | - |
+MBEDTLS_CIPHER_PADDING_ONE_AND_ZEROS | N/A | - |
+MBEDTLS_CIPHER_PADDING_ZEROS_AND_LEN | N/A | - |
+MBEDTLS_CIPHER_PADDING_ZEROS | N/A | - |
+MBEDTLS_REMOVE_ARC4_CIPHERSUITES | OPENSSL_NO_WEAK_SSL_CIPHERS | The option prevents the use of RC4 in the SSL ciphersuite but still supports the algorithm in the crypto library. OpenSSL prevents the use of RC4 as part of OPENSSL_NO_WEAK_SSL_CIPHERS option. |
+MBEDTLS_REMOVE_3DES_CIPHERSUITES | OPENSSL_NO_WEAK_SSL_CIPHERS | The option prevents the use of 3DES in the SSL ciphersuite but still supports the algorithm in the crypto library. OpenSSL prevents the use of 3DES as part of OPENSSL_NO_WEAK_SSL_CIPHERS option. |
+!MBEDTLS_ECP_DP_SECP192R1_ENABLED | N/A | Cannot be disabled on OpenSSL. |
+!MBEDTLS_ECP_DP_SECP224R1_ENABLED | N/A | Cannot be disabled on OpenSSL. |
+MBEDTLS_ECP_DP_SECP256R1_ENABLED | N/A | p256 matches NSA's suite B. OpenSSL supports SECP 256r1 (i.e., X9.62 prime256v1) by default.  |
+MBEDTLS_ECP_DP_SECP384R1_ENABLED | N/A | p384 matches NSA's suite B. OpenSSL supports SECP 384r1 by default. |
+MBEDTLS_ECP_DP_SECP521R1_ENABLED | N/A | p521 matches NSA's suite B. OpenSSL supports SPEC 521r1 by default. |
+!MBEDTLS_ECP_DP_SECP192K1_ENABLED | N/A | Cannot be disabled on OpenSSL. |
+!MBEDTLS_ECP_DP_SECP224K1_ENABLED | N/A | Cannot be disabled on OpenSSL. |
+MBEDTLS_ECP_DP_SECP256K1_ENABLED | N/A | p256k1 is used by bitcoin. OpenSSL supports SECP256k1 by default. |
+!MBEDTLS_ECP_DP_BP256R1_ENABLED | N/A | Cannot be disabled on OpenSSL. |
+!MBEDTLS_ECP_DP_BP384R1_ENABLED | N/A | Cannot be disabled on OpenSSL. |
+!MBEDTLS_ECP_DP_BP512R1_ENABLED | N/A | Cannot be disabled on OpenSSL. |
+MBEDTLS_ECP_DP_CURVE25519_ENABLED | N/A | OpenSSL supports curve 25519 by default. |
+MBEDTLS_ECP_DP_CURVE448_ENABLED | N/A | OpenSSL supports curve 448 by default. |
+MBEDTLS_ECP_NIST_OPTIM | N/A | No equivalent option in OpenSSL. |
+!MBEDTLS_ECP_RESTARTABLE | N/A | No equivalent option in OpenSSL. |
```

Comment:
This is an mbedTLS specific feature, see https://tls.mbed.org/kb/development/restartable-ecc. I would just comment that mbedTLS defaults this to disabled and so OpenSSL has parity behavior anyway by not supporting it.

Reply:
I removed this configuration as I think it's implementation specific.